docs/content/development/extensions-contrib/ganglia-emitter.md

```diff
@@ -0,0 +1,66 @@
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one
+  ~ or more contributor license agreements.  See the NOTICE file
+  ~ distributed with this work for additional information
+  ~ regarding copyright ownership.  The ASF licenses this file
+  ~ to you under the Apache License, Version 2.0 (the
+  ~ "License"); you may not use this file except in compliance
+  ~ with the License.  You may obtain a copy of the License at
+  ~
+  ~   http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+---
+layout: doc_page
+---
+
+# Ganglia Emitter
+
+To use this extension, make sure to [include](../../operations/including-extensions.html) `ganglia-emitter` extension.
+
+## Introduction
+
+This extension emits druid metrics to a Ganglia server.
+(http://ganglia.info/)
+
+## Configuration
+
+All the configuration parameters for the Ganglia emitter are under `druid.emitter.ganglia`.
+
+|property|description|required?|default|
+|--------|-----------|---------|-------|
+|`druid.emitter.ganglia.hostname`|The hostname of the Ganglia server.|yes|none|
+|`druid.emitter.ganglia.port`|The port of the Ganglia server.|yes|none|
+|`druid.emitter.ganglia.separator`|Metric name separator|no|.|
+|`druid.emitter.ganglia.includeHost`|Flag to include the hostname as part of the metric name.|no|false|
+|`druid.emitter.ganglia.dimensionMapPath`|JSON file defining the Ganglia type, and desired dimensions for every Druid metric|no|Default mapping provided. See below.|
+|`druid.emitter.ganglia.blankHolder`|The blank character replacement as Ganglia does not support path with blank character|no|"-"|
+|`druid.emitter.ganglia.flushPeriod` | Queue flushing period in milliseconds. |no|1 minute|
+|`druid.emitter.ganglia.loadPeriod` | Load metrics config period in milliseconds. |no|1 minute|
+|`druid.emitter.ganglia.maxQueueSize`| Maximum size of the queue used to buffer events. |no|`MAX_INT`|
```

Comment:
`Integer.MAX_VALUE` isn't a suitable default for the queue size. This means that if there are issues emitting to the Ganglia server, the buffer will continue to fill up until it consumes the entire heap and crashes the server.

The emitter should aim to minimize the performance impact it has on the Druid process. Attempting to limit memory usage to something like 10% of the heap would be more appropriate.

Reply:
Also, the buffer should be size based rather than number of items based for more predictable memory usage. Take a look at `MemoryBoundLinkedBlockingQueue` used by KafkaEmitter and if you can use it, it may make sense to move it to core Druid and reuse it.